source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -543,6 +578,71 @@ void CacheFilter::finalizeEncodingCachedResponse() {
   filter_state_ = FilterState::ResponseServedFromCache;
 }
 
+LookupStatus CacheFilter::lookupStatus() const {
+  if (lookup_result_ == nullptr && lookup_ != nullptr) {
+    return LookupStatus::RequestIncomplete;
+  }
+
+  if (lookup_result_ != nullptr) {
+    switch (lookup_result_->cache_entry_status_) {
+    case CacheEntryStatus::Ok:
+      return LookupStatus::CacheHit;
+    case CacheEntryStatus::Unusable:
+      return LookupStatus::CacheMiss;
+    case CacheEntryStatus::RequiresValidation:
+      // The CacheFilter sent the response upstream for validation; check the
+      // filter state to see whether and how the upstream responded. The
+      // filter currently won't send the stale entry if it can't reach the
+      // upstream or if the upstream responds with a 5xx, so don't include
+      // special handling for those cases.
+      switch (filter_state_) {
+      case FilterState::ValidatingCachedResponse:
+        return LookupStatus::RequestIncomplete;
+      case FilterState::EncodeServingFromCache:
+      case FilterState::ResponseServedFromCache:
+        return LookupStatus::StaleHitWithSuccessfulValidation;
```

Comment:
what would a caller do differently with `StaleHitWithSuccesfulValidation` as opposed to a simple `Hit`?

Reply:
The caller in this case is the metrics gathering code, reporting on how the CacheFilter is behaving, so the metrics code would report the activity. When investigating an increase in traffic to their backend, a service owner might see that there is also an increase in validations being sent from Envoy, and then realize they made a mistake in setting low TTLs in `cache-control` or other similar validation-inducing values.